src/code_memory/embeddings.py

```
@@ -0,0 +1,248 @@
+        # Optimize for inference
+        if self.device == "cuda":
+            self.model = self.model.half()  # Use FP16 for memory efficiency
+
+    def _encode_with_cache_key(
+        self, text: str, normalize: bool = True
+    ) -> NDArray[np.float32]:
+        """Encode single text with caching (cache key function)."""
+        if self.model is None:
+            raise RuntimeError("Model not initialized")
+
+        self.stats.record_cache_miss()
```

Pylint (3.3.7)

[error] Instance of 'CodeEmbedder' has no 'stats' member

(E1101)